### PR TITLE
Kruskals algorithm

### PR DESCRIPTION
**Pull request by jpfairbanks**

### CODE REVIEW

**Comment by sbromberger** [comment]
So when we get all the tests, we'll need to squash commits as well.

File: REQUIRE [hunk 1]
```diff
@@ -7,3 +7,4 @@ ParserCombinator 1.7.3
 Docile 0.5.3
 Clustering 0.4
 Combinatorics
+DataStructures
```
Comment (sbromberger) [inline REQUIRE]: Nope. This is a nonstarter. We should be trying to remove dependencies, not add new ones.

File: src/kruskal.jl [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+		error("Distances are incompatible with graph g. nv=$nv, size(distmx)=$(size(distmx))")
+	end
+	edgeList = Vector{Tuple{Float64,Int,Int}}(0)
+	for (i,j) in edges(g)
+			push!(edgeList,(distmx[i,j],i,j))
+	end
+	return edgeList
+end
+
+function Graph{T}(v::Vector{Tuple{T, Int, Int}}, n::Int)
```
Comment (sbromberger) [inline src/kruskal.jl]: This should go in `src/graph.jl`.

**Comment by hayd** [comment]
Also: please use spaces rather than tabs! It makes the indentation look whacky.

**Comment by cjamadagni** [comment]
@hayd : I'll make appropriate changes.

**Comment by sbromberger** [comment]
Is this PR still active?

**Comment by jpfairbanks** [comment]
I think we were waiting for a resolution to including DataStructures.jl
We agreed that since it is owned by JuliaLang, we can depend on it.
We just need to make sure it merges against the most recent master.

The coverage 90% is just the input validation never sees the error cases.

**Comment by sbromberger** [comment]
Very cool. BTW, you probably can (and should, if you can) test the error cases....

**Comment by sbromberger** [comment]
@jpfairbanks do you still want this PR? (If so, can we resolve conflicts?)

**Comment by sbromberger** [comment]
@jpfairbanks is there something in particular holding this up (besides the merge conflicts)?

**Comment by jpfairbanks** [comment]
Did we decide that it is ok to depend on DataStructures.jl, Or should this go in Extras?

**Comment by sbromberger** [comment]
DataStructures is 1) leaf, 2) JuliaLang, so I think it's ok. See revised https://github.com/JuliaGraphs/LightGraphs.jl/pull/277#discussion_r49953211

S.

**Comment by CarloLucibello** [comment]
closing this in favour of #465

**Pull request closed without merging**